modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/DiagramHeader.js

```diff
@@ -15,123 +15,143 @@ import ClayDropDown from '@clayui/drop-down';
-								max={100}
-								name={`${namespace}diagram-color-picker`}
-								onColorsChange={setCustomColors}
-								onValueChange={(item) => {
-									setAddNewPinState({
-										fill: item.replace('#', ''),
-										radius: addNewPinState.radius,
-									});
-								}}
-								showHex={true}
-								step={1}
-								title={Liferay.Language.get('custom-colors')}
-								useNative={newPinSettings.colorPicker.useNative}
-								value={addNewPinState.fill}
-							/>
-						</ClayDropDown.Group>
-					</ClayDropDown.ItemList>
-
-					<ClayDropDown.Divider />
+			{isAdmin && (
+				<div className="d-flex text-align-center">
+					<label className="align-middle my-auto">
+						{Liferay.Language.get('diagram-pin')}
+					</label>
 
-					<ClayDropDown.ItemList>
-						<ClayDropDown.Group
-							header={Liferay.Language.get('select-radius')}
-						>
-							{radiusChoice.map((item, i) => (
-								<ClayDropDown.Item
-									key={i}
-									onClick={() => {
+					<ClayDropDown
+						active={active}
+						className="my-auto"
+						onActiveChange={setActive}
+						trigger={
+							<ClayButton
+								alt={Liferay.Language.get(
+									'click-to-select-custom-diameter'
+								)}
+								className="ml-3 select-diameter"
+								displayType="secondary"
+							>
+								{Liferay.Language.get('default-diameter')}
+							</ClayButton>
+						}
+					>
+						<ClayDropDown.ItemList className="diagram-header-picker">
+							<ClayDropDown.Group
+								header={Liferay.Language.get('select-color')}
+							>
+								<ClayColorPicker
+									colors={customColors}
+									label={Liferay.Language.get('custom-color')}
+									max={100}
+									name={`${namespace}diagram-color-picker`}
+									onColorsChange={setCustomColors}
+									onValueChange={(item) => {
 										setAddNewPinState({
-											fill: addNewPinState.fill,
-											radius: item.value,
+											fill: item.replace('#', ''),
+											radius: addNewPinState.radius,
 										});
 									}}
-								>
-									{item.label}
-								</ClayDropDown.Item>
-							))}
-						</ClayDropDown.Group>
-					</ClayDropDown.ItemList>
+									showHex={true}
+									step={1}
+									title={Liferay.Language.get('custom-color')}
+									useNative={
+										newPinSettings.colorPicker.useNative
+									}
+									value={addNewPinState.fill}
+								/>
+							</ClayDropDown.Group>
+						</ClayDropDown.ItemList>
 
-					<ClayDropDown.Divider />
+						<ClayDropDown.Divider />
 
-					<ClayDropDown.Caption>
-						<ClayForm>
-							<ClayForm.Group className="form-group-sm">
-								<label htmlFor="slider">
-									{Liferay.Language.get('custom-radius')}
-								</label>
+						<ClayDropDown.ItemList>
+							<ClayDropDown.Group
+								header={Liferay.Language.get('select-radius')}
+							>
+								{RADIUS_CHOICE.map((item, i) => (
+									<ClayDropDown.Item
+										key={i}
```

Comment:
Instead of using numeric indexes, is there an unique identifier for each `item`? If so, I think you can use it :)